WordPress/dashicons
# Add build process for Dashicons

This PR looks bigger than it is. Mostly, it adds a Node JS package.json file so it'll be installable, as well as a Grunt file, for parsing SVG files. Much of the work is ported from Gridicons, so it outputs the same:

- minified SVG files, great for Sketch
- an SVG sprite — way of the future
- a PHP plugin — don't know if anyone ever used this, probably good to retire
- a sprite sheet in the svg-sprite folder which is copied to the docs folder so it can be used as a GH-Page

In addition to this, I moved the "unused" folder from the svg source directory into the root and renamed it svg-unused.

This PR needs a review. We need to verify that the classnames (`dashicons`, singular) is good for the sprite suggested markup. Should the React component stay? Should the PHP plugin stay?

Future improvements could include building the icon font using grunt. But this is quite an undertaking, because in order to sensure back compat with existing unicode codepoints, we'd have to manually port every unicode ID to a codepoints file, so grunt outputs the right IDs for the right icons. See https://github.com/Automattic/social-logos/blob/master/codepoints.json

Steps to test:

- make sure you have Node JS installed
- in the commandline, type `npm install`
- after the initial dependencies are installed type `npm run build`
- verify that this bakes the above mentioned

Note: the `docs/index.html` and `svg-sprite/index.html` files don't work locally, they have to run off a server of some sort due to SVG sandboxing limitations.
